### ISSUE TITLE
git status --porcelain not very performant

### ISSUE THREAD

**Issue by brejoc** [bucaran/wahoo#77]
In fact it takes ages to complete with very big repos. Perhaps inspiration could come from [zsh's vcs_info](http://zsh.sourceforge.net/Doc/Release/User-Contributions.html#Version-Control-Information) which manages to enter the directory without notable delay.

Steps to reproduce:

1. Start fish with e.g. the batman theme.
2. Navigate to a very big git repo.
3. Wait 20 seconds for fish to accept user input again.

**Comment by brejoc** [comment]
I think I've been a bit hasty here. The zsh-config I've been using didn't show me if there where changes in the repo and that what `git status --porcelain` is used for here, isn't it?

**Comment by bucaran** [comment]
@brejoc Thanks for bringing this up. I just noticed `batman` is not using the `git` functions built in Wahoo (See [here](https://github.com/wa/wahoo/tree/master/lib/git)). It should as it helps reducing code duplication and I just created an issue for that.

Now, I looked at the git functions, and `git_is_touched` looks like this:

```fish
function git_is_touched -d "Check if repo has any changes"
  git_is_repo; and begin
    test -n (echo (command git status --porcelain))
  end
end
```

What would be a faster way to check if the repo is touched / dirty?
